src/trading/NewOrdersFetcher.sol

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+ */
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/libraries/arrays/Bytes20Arrays.sol';
+import 'ROOT/libraries/Trading.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/NewOrders.sol';
+
+
+/**
+ * @title OrdersFetcher
+ */
+// FIXME: This was named NewOrdersFetcher to prevent naming conflicts with ordersFetcher.se. Should be renamed to OrdersFetcher.
+contract NewOrdersFetcher is Controlled {
+    using Bytes20Arrays for bytes20[];
+
+    event Log(uint256 fxpPrice);
+    
+    /**
+     * @dev Get orders for a particular market, type, and outcome (chunked)
+     */
+    function getOrderIds(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, bytes20 _startingOrderId, uint256 _numOrdersToLoad) public constant returns (bytes20[]) {
+        require(_type == Trading.TradeTypes.Bid || _type == Trading.TradeTypes.Ask);
+        require(0 <= _outcome && _outcome < _market.getNumberOfOutcomes());
+        NewOrders _orders = NewOrders(controller.lookup("NewOrders"));
+        if (_startingOrderId == 0) {
+            _startingOrderId = _orders.getBestOrderId(_type, _market, _outcome);
+        }
+        bytes20[] _orderIds;
+        _orderIds[0] = _startingOrderId;
+        uint256 _i = 0;
+        while (_i < _numOrdersToLoad && _orders.getWorseOrderId(_orderIds[_i], _type, _market, _outcome) != 0) {
```

Comment:
Change this to a for loop, since that is really what it is.  the `_orders.getWorseOrderId(...) != 0` can be changed to a `if (...) { break; }` statement inside the loop.